### PR TITLE
Remove email parser overhead on search

### PR DESCRIPTION
**Pull request by vladsf**
Loading a full email body, parsing headers, attachments is
irrelevant for the search results. This change improves speed
and eliminates possible exceptions thrown on a full email load.

IMPORTANT:
As a contributor I must add Oracle Copyright notice to the place designated for such notices (AUTHORS).

### CODE REVIEW

**Comment by glensc** [comment]
I'm sorry, what about Oracle, care to elaborate?

**Comment by glensc** [comment]
Please add Changelog entry, so it would be in place with a merge.

File: lib/eventum/class.support.php [hunk 1]
```diff
@@ -1451,6 +1451,34 @@ public static function getEmailDetails($sup_id)
         return $res;
     }
 
+    /**
+     * Method used to get the support email body and subject.
+     *
+     * @param   int $sup_id The support email ID
+     * @return  array The email entry summary
+     */
+    public static function getEmailSummary($sup_id)
```
Comment (glensc) [inline lib/eventum/class.support.php]: please add php type hints

**Comment by glensc** [comment]
btw, you'll likely find this one also huge performance boost: https://github.com/eventum/eventum/pull/485

File: lib/eventum/class.support.php (hunk 1 above)
Comment (vladsf) [reply lib/eventum/class.support.php]: type hint added.

File: lib/eventum/class.support.php [hunk 2]
```diff
@@ -1451,6 +1451,34 @@ public static function getEmailDetails($sup_id)
         return $res;
     }
 
+    /**
+     * Method used to get the support email body and subject.
+     *
+     * @param   int $sup_id The support email ID
+     * @return  array The email entry summary
+     */
+    public static function getEmailSummary($sup_id): array
```
Comment (glensc) [inline lib/eventum/class.support.php]: type hint for parameter as well, this branch requires php 7.1

**Comment by vladsf** [comment]
> I'm sorry, what about Oracle, care to elaborate?
even though MySQL and Sun are already there, this new Oracle notice is a requirement (for me as a contributor affiliated with Oracle) for pull requests sent to an open source project. You may remove it, but I guess I will not be able to send bug fixes, minor enhancements afterwards. Only bug reports.

**Comment by glensc** [comment]
3c5925487cf08b8727bddd838883fc82b1f1209e

**Pull request merged**